File: japiwznm/WznmQSteAStep.java
/**
  * \file WznmQSteAStep.h
  * Java API code for record of table TblWznmQSteAStep
  * \author Alexander Wirthmueller
  * \date created: 11 Jul 2020
  * \date modified: 11 Jul 2020
  */

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQSteAStep {
	
	public WznmQSteAStep(
				int jnum
				, String stubSnxRefWznmMState
				, String srefIxVTrigger
				, String titIxVTrigger
				, String stubRefWznmMRtjob
				, String stubRefWznmMVectoritem
				, String xsref
				, String stubRefWznmMRtdpch
				, String srefsMask
				, String Cond
				, String yesnoCustcode
			) {
		this.jnum = jnum;
		this.stubSnxRefWznmMState = stubSnxRefWznmMState;
		this.srefIxVTrigger = srefIxVTrigger;
		this.titIxVTrigger = titIxVTrigger;
		this.stubRefWznmMRtjob = stubRefWznmMRtjob;
		this.stubRefWznmMVectoritem = stubRefWznmMVectoritem;
		this.xsref = xsref;
		this.stubRefWznmMRtdpch = stubRefWznmMRtdpch;
		this.srefsMask = srefsMask;
		this.Cond = Cond;
		this.yesnoCustcode = yesnoCustcode;
	};

	public int jnum;
	public String stubSnxRefWznmMState;
	public String srefIxVTrigger;
	public String titIxVTrigger;
	public String stubRefWznmMRtjob;
	public String stubRefWznmMVectoritem;
	public String xsref;
	public String stubRefWznmMRtdpch;
	public String srefsMask;
	public String Cond;
	public String yesnoCustcode;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQSteAStep");

		if (Xmlio.checkXPath(doc, basexpath)) {
			stubSnxRefWznmMState = Xmlio.extractStringUclc(doc, basexpath, "stubSnxRefWznmMState", "snx", null, 0);
			srefIxVTrigger = Xmlio.extractStringUclc(doc, basexpath, "srefIxVTrigger", "trg", null, 0);
			titIxVTrigger = Xmlio.extractStringUclc(doc, basexpath, "titIxVTrigger", "trg2", null, 0);
			stubRefWznmMRtjob = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMRtjob", "rtj", null, 0);
			stubRefWznmMVectoritem = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMVectoritem", "vit", null, 0);
			xsref = Xmlio.extractStringUclc(doc, basexpath, "xsref", "xsr", null, 0);
			stubRefWznmMRtdpch = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMRtdpch", "rtd", null, 0);
			srefsMask = Xmlio.extractStringUclc(doc, basexpath, "srefsMask", "msk", null, 0);
			Cond = Xmlio.extractStringUclc(doc, basexpath, "Cond", "cnd", null, 0);
			yesnoCustcode = Xmlio.extractStringUclc(doc, basexpath, "yesnoCustcode", "ccd", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwznm/WznmQSteAStep.cpp
/**
	* \file WznmQSteAStep.cpp
	* API code for table TblWznmQSteAStep (implementation)
	* \author Alexander Wirthmueller
	* \date created: 11 Jul 2020
	* \date modified: 11 Jul 2020
	*/

#include "WznmQSteAStep.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQSteAStep
 ******************************************************************************/

WznmQSteAStep::WznmQSteAStep(
			const uint jnum
			, const string stubSnxRefWznmMState
			, const string srefIxVTrigger
			, const string titIxVTrigger
			, const string stubRefWznmMRtjob
			, const string stubRefWznmMVectoritem
			, const string xsref
			, const string stubRefWznmMRtdpch
			, const string srefsMask
			, const string Cond
			, const string yesnoCustcode
		) {
	this->jnum = jnum;
	this->stubSnxRefWznmMState = stubSnxRefWznmMState;
	this->srefIxVTrigger = srefIxVTrigger;
	this->titIxVTrigger = titIxVTrigger;
	this->stubRefWznmMRtjob = stubRefWznmMRtjob;
	this->stubRefWznmMVectoritem = stubRefWznmMVectoritem;
	this->xsref = xsref;
	this->stubRefWznmMRtdpch = stubRefWznmMRtdpch;
	this->srefsMask = srefsMask;
	this->Cond = Cond;
	this->yesnoCustcode = yesnoCustcode;
};

bool WznmQSteAStep::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQSteAStep");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "stubSnxRefWznmMState", "snx", stubSnxRefWznmMState);
		extractStringUclc(docctx, basexpath, "srefIxVTrigger", "trg", srefIxVTrigger);
		extractStringUclc(docctx, basexpath, "titIxVTrigger", "trg2", titIxVTrigger);
		extractStringUclc(docctx, basexpath, "stubRefWznmMRtjob", "rtj", stubRefWznmMRtjob);
		extractStringUclc(docctx, basexpath, "stubRefWznmMVectoritem", "vit", stubRefWznmMVectoritem);
		extractStringUclc(docctx, basexpath, "xsref", "xsr", xsref);
		extractStringUclc(docctx, basexpath, "stubRefWznmMRtdpch", "rtd", stubRefWznmMRtdpch);
		extractStringUclc(docctx, basexpath, "srefsMask", "msk", srefsMask);
		extractStringUclc(docctx, basexpath, "Cond", "cnd", Cond);
		extractStringUclc(docctx, basexpath, "yesnoCustcode", "ccd", yesnoCustcode);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQSteAStep
 ******************************************************************************/

ListWznmQSteAStep::ListWznmQSteAStep() {
};

ListWznmQSteAStep::ListWznmQSteAStep(
			const ListWznmQSteAStep& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQSteAStep(*(src.nodes[i]));
};

ListWznmQSteAStep& ListWznmQSteAStep::operator=(
			const ListWznmQSteAStep& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQSteAStep(*(src.nodes[i]));

	return *this;
};

ListWznmQSteAStep::~ListWznmQSteAStep() {
	clear();
};

void ListWznmQSteAStep::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQSteAStep::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQSteAStep* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQSteAStep");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQSteAStep", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQSteAStep(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQSteAStep[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

